Formula/tcpreplay.rb

```diff
@@ -17,17 +17,36 @@ class Tcpreplay < Formula
   depends_on "libtool" => :build
   depends_on "libdnet"
 
+  uses_from_macos "libpcap"
+
   def install
-    ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version
     args = %W[
       --disable-debug
       --disable-dependency-tracking
       --disable-silent-rules
       --prefix=#{prefix}
       --enable-dynamic-link
-      --with-macosx-sdk=#{MacOS.version}
     ]
 
+    on_macos do
+      ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version
+      args << "--with-macosx-sdk=#{MacOS.version}"
+
+      # The SDK is currently found using `xcrun --sdk macosx<V>` starting with
+      # input `--with-macosx-sdk=<V>` and then going from older 10.8 onward.
+      # On ARM, for Big Sur 11.4 the correct SDK is 11.3 (as of 2021-07-11);
+      # however, the logic picks 10.15, which causes configure failure.
+      # As a workaround, we remove all 10.x versions from SDK detection logic.
+      #
+      # Check in next release if the workaround can be removed.
+      # Upstream issue: https://github.com/appneta/tcpreplay/issues/668
+      inreplace "configure.ac", /(\$with_macosx_sdk[ \t]+)(?:10\.\d+[ \t]+)+/, "\\1" if Hardware::CPU.arm?
```

Comment:
```suggestion
      inreplace "configure.ac", /(\$with_macosx_sdk\s+)(?:10\.\d+\s+)+/, "\\1" if Hardware::CPU.arm?
```
Isn't this (mostly) equivalent? Or do we want to avoid matching the extra characters in `\s`?

The error here probably needs to be reported upstream.

Reply:
Should be fine to switch if we want to use this workaround.

There is also the other option I mentioned of detecting value, but will require extracting value from command-line output, which I don't know how stable it is. I don't know best Formula syntax, but maybe something like:
```rb
      sdk_version = shell_output("xcrun --show-sdk-version 2>/dev/null").chomp
      args << "--with-macosx-sdk=#{sdk_version}"
```

Right now `--with-macosx-sdk=#{MacOS.version}` is a no-op on Big Sur since "11" is not a valid input:
```console
❯ sudo xcode-select -s /Library/Developer/CommandLineTools

❯ xcrun --sdk macosx11 --show-sdk-path
xcrun: error: SDK "macosx11" cannot be located
xcrun: error: SDK "macosx11" cannot be located
xcrun: error: unable to lookup item 'Path' in SDK 'macosx11'
```

Reply:
Added change into PR and should be good to merge now.
Workaround should be fine until upstream fixes: https://github.com/appneta/tcpreplay/issues/668#issuecomment-879207337

It looks like plan is to update `configure.ac` to properly deal with CLT and to support user-provided minimum versions with a new argument like `--with-macos-min-version`.